Tasks/PowerShellOnTargetMachines2/task.json

```diff
@@ -0,0 +1,201 @@
+    "description": "Execute PowerShell scripts on remote machine(s)",
+    "helpMarkDown": "[More Information](https://go.microsoft.com/fwlink/?linkid=627414)",
+    "category": "Deploy",
+    "visibility": [
+        "Build",
+        "Release"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 3,
+        "Minor": 0,
+        "Patch": 23
+    },
+    "preview": true,
+    "groups": [
+        {
+            "name": "ScriptOptions",
+            "displayName": "Script options",
+            "isExpanded": true
+        },
+        {
+            "name": "SessionOptions",
+            "displayName": "PSSession options",
+            "isExpanded": true
+        },
+        {
+            "name": "ErrorHandlingOptions",
+            "displayName": "Error handling options",
+            "isExpanded": true
+        },
+        {
+            "name": "advanced",
+            "displayName": "Advanced",
+            "isExpanded": false
+        }
+    ],
+    "releaseNotes": "What's new in Version 3.0: <br/>&nbsp;&nbsp; Use invoke-command to perform remote script execution",
+    "minimumAgentVersion": "2.0.0",
+    "inputs": [
+        {
+            "name": "Machines",
+            "type": "multiLine",
+            "label": "Machines",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Provide a comma separated list of machine IP addresses or FQDNs along with ports. Port is defaulted based on the selected protocol. <br>Eg: dbserver.fabrikam.com,dbserver_int.fabrikam.com:5986,192.168.12.34:5986 <br>Or provide output variable of other tasks. Eg: $(variableName) <br>If you are using HTTPS, name/IP of machine should match the CN in the certificate."
+        },
+        {
+            "name": "UserName",
+            "type": "string",
+            "label": "Admin Login",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Administrator login for the target machines. <br> Eg: Format: Domain\\Admin User, Admin User@Domain, .\\Admin User"
+        },
+        {
+            "name": "UserPassword",
+            "type": "string",
+            "label": "Password",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "Administrator password for the target machines. <br>It can accept variable defined in Build/Release definitions as '$(passwordVariable)'. <br>You may mark variable type as 'secret' to secure it."
+        },
+        {
+            "name": "ScriptType",
+            "type": "radio",
+            "label": "Script Type",
+            "defaultValue": "FilePath",
+            "options": {
+                "FilePath": "File Path",
+                "Inline": "Inline"
+            },
+            "helpMarkDown": "The type of script to execute: Inline or File Path",
+            "groupName": "ScriptOptions"
+        },
+        {
+            "name": "ScriptPath",
+            "type": "string",
```

Comment:
should it be type filePath

Reply:
this path needs to be on the target machine. changing its type to filePath will cause people to give paths as they appear in currently linked artifact (i.e. relative to the agent machine). I think this will cause confusion as they might expect that we do the copying over of the files .